scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1258,6 +1267,7 @@ def test_align_vectors_noise():
     n_vectors = 100
     rot = Rotation.random(random_state=rnd)
     vectors = rnd.normal(size=(n_vectors, 3))
+    vectors /= np.linalg.norm(vectors, axis=1).reshape(n_vectors, 1)
```

Comment:
@nmayorov I added a normalization step in `align_vectors` because I believe the calculation of `rssd` should not be dependent on the vector length (if a set of vectors of varying length can align exactly, I would expect rssd to be 0). Let me know if I'm wrong on that

Reply:
As far as I remember, originally it was done in this method (with a flag whether vectors are already normalized).

However after some discussion this logic was removed and I think it's a correct decision. The reason is that we don't need to have normalized vectors to have a well defined problem. Yes, if vectors are not normalized even with perfect (angular) alignment the cost function is not zero. However this is a transparent behavior which can be completely avoided by normalizing the vectors beforehand. The decision of whether or not normalize vectors are up to the user.

At this point, I think, it's certainly a very undesirable to change this behavior. 

It doesn't seem to create any logical issues for the extensions you are working on, does it?

Reply:
I don't quite understand the use case where you want rssd weighted by vector length, but I agree that it's more general. I removed the normalization step and added some information to the docstring for `rssd` to make the behavior clear.

Reply:
Originally I though the same. But if there is a need to align vectors of different lengths (i.e. implicit weighing), why not provide this option. And then it's better not to complicate things by adjusting `rssd` definition.